### ISSUE TITLE
[2.5] Breakout broadcast messages incomplete if `bufferChatInsertsMs` > 0

### ISSUE THREAD

**Issue by timm2k** [bigbluebutton/bigbluebutton#15185]
**Describe the bug**
Hi guys,

If I set `public.chat.bufferChatInsertsMs 250` in /etc/bigbluebutton/bbb-html5.yml, broadcast messages to breakoutrooms did not appear in all rooms as expected.

**To Reproduce**
1. install BBB 2.5.1
2. `yq w -i  /etc/bigbluebutton/bbb-html5.yml public.chat.bufferChatInsertsMs 250 && bbb-conf --restart`
3. start meeting
4. join with user A and B
5. create two breakout rooms and place one user in each breakout room
6. moderator sends multiple broadcast messages like "1, 2, 3, 4, 5, 6, 7, 8, 9" 
7. not all messages were received in all breakout rooms

**Expected behavior**
All broadcast messages are viewed in all breakout rooms of the meeting.

**Actual behavior**
Acually for me it is random which message is shown in a breakout room.

**Screenshots**
![image](https://user-images.githubusercontent.com/5180299/173811782-cd21f0ee-c903-418d-a7a7-1a452ada8960.png)
(Screenshot was taken with `bufferChatInsertsMs: 100`)

**Workaround**
Remove `bufferChatInsertsMs` from `/etc/bigbluebutton/bbb-html5.yml`:
`yq d -i  /etc/bigbluebutton/bbb-html5.yml public.chat.bufferChatInsertsMs`

**BBB version:**
BigBlueButton Server 2.5.1 (3020)

**Comment by defnull** [comment]
Just a very wild guess after skimming code for a while:
- Chat messages have a timestamp, and multiple consecutive messages from the same sender and within the same rounded second are grouped together and displayed differently.
- Setting `bufferChatInsertsMs` increases the change that a message arrives in a different wall-clock second than their timestamp says. Which means, two messages that should be grouped together by timestamp would not be grouped together by wall-clock arrival-time.
- Another idea would be that the `insertMany` triggered by `bufferChatInsertsMs` results in a different ordering of these events, confusing the message grouper.

I did not find the actual bug yet, but it might be within the message grouping or display logic, not within the actual event processing.

**Comment by defnull** [comment]
After staring at code for a bit longer, I think that the second theory is more likely. Message buffering uses `insertMany` (see `addBulkGroupChatMsgs.js`) instead of inserting individual messages and also generates document IDs in advanced. This may result in a different 'natural ordering' within the mongodb collection.

In `bigbluebutton-html5/imports/ui/components/components-data/chat-context/context.jsx` I cannot find any code that ensures a strict timestamp ordering, so `generateStateWithNewMessage` may see events out of order and the grouping logic may get confused by non-monotonic timestamps. Still not sure, the meteor code is really hard to follow, but that's my best guess.
